src/xds/metrics.rs

```diff
@@ -58,3 +85,18 @@ impl Recorder<ConnectionTerminationReason, u64> for Metrics {
             .inc_by(count);
     }
 }
+
+impl Recorder<DeltaDiscoveryResponse, ()> for Metrics {
+    fn record(&self, response: &DeltaDiscoveryResponse, _: ()) {
+        let type_url = TypeUrl {
+            url: response.type_url.clone(),
+        };
+        self.message_types.get_or_create(&type_url).inc_by(1);
+
+        let mut message_size: usize = 0;
+        for resource in &response.resources {
+            message_size += resource.resource.as_ref().unwrap().value.len();
```

Comment:
unwrap here is not safe, we should just skip if its empty

```suggestion
            message_size += resource.resource.as_ref().map(|v| v.value.len()).unwrap_or_default();
```